### PR TITLE
fix: Forward getId to useTabsModel to ensure custom ids work when registering items

### LINKED ISSUE #4140: `getId` on `useTabsModel` wasn't forwarding correctly so custom `id`s didn't work
## 🐛 Bug Report

When having items like :
```
 const [items] = React.useState<MyTabItem[]>([
    {contextId: 'first', text: 'First Tab', contents: 'Contents of First Tab'},
    {contextId: 'second', text: 'Second Tab', contents: 'Contents of Second Tab'},
    {contextId: 'third', text: 'Third Tab', contents: 'Contents of Third Tab'},
    {contextId: 'fourth', text: 'Fourth Tab', contents: 'Contents of Fourth Tab'},
    {contextId: 'fifth', text: 'Fifth Tab', contents: 'Contents of Fifth Tab'},
    {contextId: 'sixth', text: 'Sixth Tab', contents: 'Contents of Sixth Tab'},
    {contextId: 'seventh', text: 'Seventh Tab', contents: 'Contents of Seventh Tab'},
  ]);
```

Using `getId` on the `useTabsModel` or on the element threw the error that `getId` wasn't passed to the model for the collection system to register those `id`s. We didn't catch this because our storybook examples used `id` which works with the collection system, but we should allow custom `id` properties.

### PR DESCRIPTION
**Pull request by mannycarrera4**
## Summary

Fixes: https://github.com/Workday/canvas-kit/issues/4140

## Release Category
Components

---

## Checklist

- [ ] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--docs)
- [ ] Label `ready for review` has been added to PR

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)
- [ ] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [ ] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Areas for Feedback? (optional)

- [ ] Code
- [ ] Documentation
- [ ] Testing
- [ ] Codemods

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved tab menu behavior by preserving overflow items, filtering non-interactive items, and maintaining selection when the menu opens.
  * Added reliable fallback handling for tab identifiers and display text without overriding custom menu settings.
  * Updated overflow tab identification for more consistent tab selection.
  * Fixed overflow handling so the selected tab remains visible with custom identifiers and incomplete item measurements.
  * Improved stability when item lists are empty or identifiers overlap with inherited properties.

### CODE REVIEW

File: modules/react/tabs/lib/useTabsModel.tsx [hunk 1]
```diff
@@ -110,21 +110,26 @@ export const useTabsModel = createModelHook({
...
-      },
-    })
-  );
+  const mergedMenuConfig = useMenuModel.mergeConfig(config.menuConfig, {
+    id: `menu-${model.state.id}`,
+    items: overflowItems,
+    // `getId`/`getTextValue` aren't callbacks or guards, so `mergeConfig` would let these
+    // unconditionally clobber a `menuConfig.getId`/`getTextValue` override. Fall back to the
+    // top-level Tabs config only when the caller hasn't set one on `menuConfig` directly.
+    getId: config.menuConfig?.getId || getId,
```
Comment (mannycarrera4) [inline modules/react/tabs/lib/useTabsModel.tsx:119]: this is the real change, this was returning undefined

File: modules/react/tabs/stories/examples/OverflowTabs.tsx [hunk 2]
```diff
@@ -7,23 +7,24 @@ import {px2rem} from '@workday/canvas-kit-styling';
...
+    {contextId: 'second', text: 'Second Tab', contents: 'Contents of Second Tab'},
+    {contextId: 'third', text: 'Third Tab', contents: 'Contents of Third Tab'},
+    {contextId: 'fourth', text: 'Fourth Tab', contents: 'Contents of Fourth Tab'},
+    {contextId: 'fifth', text: 'Fifth Tab', contents: 'Contents of Fifth Tab'},
+    {contextId: 'sixth', text: 'Sixth Tab', contents: 'Contents of Sixth Tab'},
+    {contextId: 'seventh', text: 'Seventh Tab', contents: 'Contents of Seventh Tab'},
   ]);
   const model = useTabsModel({
     items,
+    getId: (item: MyTabItem) => item.contextId,
```
Comment (mannycarrera4) [inline modules/react/tabs/stories/examples/OverflowTabs.tsx:27]: Changed this story to match a real work example

File: modules/react/tabs/lib/useTabsModel.tsx (hunk 1 above)
Comment (sheelah) [inline modules/react/tabs/lib/useTabsModel.tsx:119]: Do we want to add a Cypress test with a custom getId & overflow menu scenario?

**Review by sheelah — APPROVED** [review]

File: modules/react/tabs/lib/useTabsModel.tsx (hunk 1 above)
Comment (mannycarrera4) [reply modules/react/tabs/lib/useTabsModel.tsx:119]: I update the story and we have a test for OverFlowTabs in cypress, so by changing the story, it should still pass!

Comment (sheelah) [reply modules/react/tabs/lib/useTabsModel.tsx:119]: Ah ok. I'm still trying to figure out when we add tests for specific scenarios in Cypress tests vs just having them in Storybook.

Comment (mannycarrera4) [reply modules/react/tabs/lib/useTabsModel.tsx:119]: Its confusing sometimes we use storybook for cypress
